pulp_rpm/app/downloaders.py

```diff
@@ -82,11 +81,15 @@ async def _run(self, extra_data=None):
         This method provides the same return object type and documented in
         :meth:`~pulpcore.plugin.download.BaseDownloader._run`.
         """
+        # Some upstream-repos (eg, Amazon) require url-encoded paths for things like "libc++"
+        # Let's make them happy
+        new_url = quote(unquote(self.url), safe=":/")
```

Comment:
LGTM.  Can we trigger a test run that includes *all* the repos? (the ones that are normally skipped due to how long they take - and not necessarily before the merge but prior to the next release, maybe overnight tomorrow?)

Reply:
Hm. The problem is, one needs 'immediate' syncs to travel this code-path, and disk-space would become a **major** issue. Also, the "6 hour run" of all the test_fips repos is on_demand - adding the xfer-time for the content would balloon that out wildly.

What I can do, is set up a test that syncs a fedora, rhel, suse, centos, and amazon repo, deleting the repo and running orphan-cleanup in between each. We'll see how that goes.